tech-far-hub/gatsby-node.ts

```diff
@@ -0,0 +1,86 @@
+import type { GatsbyNode } from "gatsby";
+import path from "path";
+
+interface ITemplatedNode {
+  id: string;
+  parent: {
+    id: string;
+    name: string;
+    base: string;
+    relativeDirectory: string;
+    relativePath: string;
+  };
+  frontmatter: {
+    slug?: string;
+  };
+  internal: {
+    contentFilePath: string;
+  };
+}
+
+interface ITemplateNodeResultSet {
+  allMdx: {
+    edges: ReadonlyArray<{ readonly node: ITemplatedNode }>;
```

Comment:
ReadonlyArray that's a new one for me! Nice!

Reply:
I learned it from the auto-generated types